x-pack/plugins/observability/public/components/shared/exploratory_view/hooks/use_add_to_case.test.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useAddToCase } from './use_add_to_case';
+import React, { useEffect } from 'react';
+import { render } from '../rtl_helpers';
+import { EuiButton } from '@elastic/eui';
+import { fireEvent } from '@testing-library/dom';
+
+describe('useAddToCase', function () {
+  function setupTestComponent() {
+    const setData = jest.fn();
+    function TestComponent() {
+      const getToastText = jest.fn();
+
+      const result = useAddToCase({
+        lensAttributes: { title: 'Test lens attributes' } as any,
+        timeRange: { to: 'now', from: 'now-5m' },
+        getToastText,
+      });
+      useEffect(() => {
+        setData(result);
+      }, [result]);
+
+      return (
+        <span>
+          <EuiButton onClick={result.goToCreateCase}>Add new case</EuiButton>
+          <EuiButton onClick={() => result.onCaseClicked({ id: 'test' } as any)}>
+            On case click
+          </EuiButton>
+        </span>
+      );
+    }
+
+    const renderSetup = render(<TestComponent />);
+
+    return { setData, ...renderSetup };
+  }
+  it('should return expected result', async function () {
+    const { setData, core, findByText } = setupTestComponent();
+
+    expect(setData).toHaveBeenLastCalledWith({
```

Comment:
Is there any reason you can't use `renderHook` to test this, instead of creating your own helper component?

To me it seems this is partially testing your helper component, since set data and when/how it's called isn't part of the actual hook. It seems like you're wanting to just test the `result` object. 

Using `renderHook` you could pass the appropriate hook props and test that the result `object` matches your expectation.

Reply:
i need contexts on top of hook render for this to work, render from rtl utility already has those context as wrapper in. 

for example that hook uses `useKibana` inside, if i wanted to use renderHook, i had to mock all those hooks separately.

Reply:
Not sure if this fits your use case, but you can also use the `wrapper` option to wrap the hooks in providers. Could take one of the provider wrapper in our rtl helpers and wrap it with. But I see what you did here. 

https://react-hooks-testing-library.com/usage/advanced-hooks